src/store/reducers/note.reducer.ts

```diff
@@ -22,7 +23,7 @@ const noteReducer = createReducer(initialState, (builder) => {
     .addCase(noteActions.getNotesSuccess, (state, action) => {
       state.notesLoading = false;
       if (action.payload) {
-        state.notes = (action.payload as NoteInterface[]).reverse();
+        state.notes = (action.payload as NoteInterface[]).filter((note) => !note.deleted).reverse();

```

Comment:
No filterting in the reducer to let the store contain all of the notes.
Filtering is already done at the component level.